src/main/webapp/script.js

```diff
@@ -58,16 +58,23 @@ function displayPotentialMatchInfo(pmID) {
     const name = userinfo.name;
     const bio = userinfo.bio;
     const carouselContainer = document.getElementById("carousel-inner");
-    const numPhotos = 5; //hard coded value for now, once blobstore works, this will be the amount of images the user has uploaded
-    for (let i = 0; i < numPhotos; i++) {
+    let numPhotos = 0;
+    for (let i = 0; i < userinfo.blobkeys.length; i++) {
       let slideshowElement;
-      if (i === 0) {
-        slideshowElement = createSlideshowElement("images/noBlobStoreImage.jpg", "carousel-item active", name, bio);
+      if (i === 0 && userinfo.blobkeys[i] != "") {
+        numPhotos++;
+        slideshowElement = createSlideshowElement(userinfo.blobkeys[i], "carousel-item active", name, bio, false);
+        carouselContainer.appendChild(slideshowElement);
       }
-      else {
-        slideshowElement = createSlideshowElement("images/noBlobStoreImage.jpg", "carousel-item", name, bio);
+      else if (userinfo.blobkeys[i] != "") {
+        numPhotos++;
+        slideshowElement = createSlideshowElement(userinfo.blobkeys[i], "carousel-item", name, bio, false);
+        carouselContainer.appendChild(slideshowElement);
```

Comment:
Looks like the only difference between this and the preceding if() is the class name? If so, can simplify to

```js
if (userinfo.blobkeys[i] != "") {
  numPhotos++;
  slideshowElement = createSlideshowElement(userinfo.blobkeys[i], "carousel-item" + (i === 0 ? " active" : ""), name, bio, false);
  carouselContainer.appendChild(slideshowElement);
}
```